.github/workflows/checks.yml

```diff
@@ -0,0 +1,160 @@
+name: Checks
+
+on:
+  push:
+    branches:
+      - master
+      - cloud
+      - "release-*"
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  check-app-layers:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Extract version of Go to use
+      run: echo "GOVERSION=$(cat go.mod | grep go | head -1 | awk ' { print $2 } ' | cut -d '@' -f 1 | cut -d ':' -f 2)" >> $GITHUB_ENV
+
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GOVERSION }}
+
+    - name: Run check-app-layers
+      run: |
+        make app-layers
+        if [[ -n $(git status --porcelain) ]]; then echo "Please update the app layers using 'make app-layers'"; exit 1; fi
+
+  check-store-layers:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Extract version of Go to use
+      run: echo "GOVERSION=$(cat go.mod | grep go | head -1 | awk ' { print $2 } ' | cut -d '@' -f 1 | cut -d ':' -f 2)" >> $GITHUB_ENV
+
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GOVERSION }}
+
+    - name: Run check-store-layers
+      run: |
+        make store-layers
+        if [[ -n $(git status --porcelain) ]]; then echo "Please update the store layers using 'make store-layers'"; exit 1; fi
+
+  check-mocks:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Extract version of Go to use
+      run: echo "GOVERSION=$(cat go.mod | grep go | head -1 | awk ' { print $2 } ' | cut -d '@' -f 1 | cut -d ':' -f 2)" >> $GITHUB_ENV
+
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GOVERSION }}
+
+    - name: Run check-mocks
+      run: |
+        make einterfaces-mocks store-mocks
+        if [[ -n $(git status --porcelain) ]]; then echo "Please update the mocks using 'make einterfaces-mocks store-mocks'"; exit 1; fi
+
+  check-migrations:
+    runs-on: ubuntu-latest
```

Comment:
This does not seem to run on the mattermost docker image we have.

Reply:
in circleci there is no easy way to install go or have that available in the env, that is why we run over a container.
we can do that here if you want as well